### ISSUE TITLE
sox output to be streamed 

### ISSUE THREAD

**Issue by muammar** [StreamMachine/StreamMachine#86]
I am running locally streammachine as shown in the **Getting started** section, and I have the following doubt. I am capturing the Mac OS X core audio using _Soundflower_ and _sox_ as shown below:

`sox -t coreaudio "Soundflower (2ch)" -q /tmp/stream.mp3`

Then, I am trying to stream the `stream.mp3` as follows:

`./streammachine-util-cmd source --port 8002 --stream test --password testing /tmp/stream.mp3`

Even if everything seems to work as expected, only one fragment of the mp3 that is being constantly populated is played when I go to `http://localhost:8001/test`. Is this the expected result or not?. I would have expected the file to keep playing as it gets more data from coreaudio. 

Thank you very much for this great server.

**Comment by ewr** [comment]
The source utility is only intended to be used for static files in development. Basically it reads the file and then sends its contents to the server in a loop. Great for a testing source, but not so much for your use.

You might check out [Ladiocast](https://itunes.apple.com/us/app/ladiocast/id411213048?mt=12) or [butt](http://butt.sourceforge.net) as source options for the Mac.

If you look at util/icecast_source, though, you could also put together your own tool pretty simply:

https://github.com/StreamMachine/StreamMachine/blob/master/src/streammachine/util/icecast_source.coffee

Basically you just need to send the request headers and then start piping audio after you see the 200 response.
